### ISSUE TITLE
String chunk reader

### ISSUE THREAD

**Issue by copy** [rlepigre/ocaml-imagelib#6]
A chunk reader for strings would be useful when working with images from a script that dumps the image to stdout, so one doesn't have to mess with files and can keep everything in memory.

I made an implementation, but a short review would be helpful (I'm not entirely sure what to return in the `` `Close`` case and in the `` `End_of_file`` constructor):

```ocaml
let chunk_reader_of_string s =
  let index = ref 0 in
  function
  | `Close -> Ok ""
  | `Bytes n ->
    if n + !index > String.length s then
      Error (`End_of_file (String.length s))
    else (
      let old_index = !index in
      index := !index + n;
      Ok (String.sub s old_index n)
    )
```

**Comment by cfcs** [comment]
Good idea! I rewrote it a little bit to make it easier to read, and submitted a PR with your function (see link above).

The `Close` case is correct; it's mostly a stub to enable cleaning up side-effect readers like file descriptors that need to be freed after use. 

`End_of_file`: Also correct. The position is hidden by `get_bytes` anyway. When I wrote it I thought it might be helpful to know where the end was, and how many bytes we expected after the end, but I did not want to change the exception interface too much, so I didn't expose these integers in `exception End_of_file`.
